src/precompiles/mod.rs

```diff
@@ -78,7 +70,12 @@ pub trait Precompile {
     fn required_gas(input: &[u8]) -> Result<u64, ExitError>;
 
     /// Runs the precompile function.
-    fn run(input: &[u8], target_gas: u64, context: &Context, is_static: bool) -> PrecompileResult;
+    fn run(
+        input: &[u8],
+        target_gas: Option<u64>,
+        context: &Context,
+        is_static: bool,
+    ) -> EvmPrecompileResult;
```

Comment:
Why do we make `target_gas` optional? I see that this is present in the upstream as well, but what's the purpose?

Reply:
I've never figured that out to be fair. That could change in the future. For now, its just always something.